src/Http/Headers/src/EntityTagHeaderValue.cs

```diff
@@ -151,36 +152,36 @@ public bool Compare(EntityTagHeaderValue other, bool useStrongComparison)
         public static EntityTagHeaderValue Parse(StringSegment input)
         {
             var index = 0;
-            return SingleValueParser.ParseValue(input, ref index);
+            return SingleValueParser.ParseValue(input, ref index)!;
         }
 
         public static bool TryParse(StringSegment input, out EntityTagHeaderValue parsedValue)
```

Comment:
Shouldn't this be the following?

```suggestion
        public static bool TryParse(StringSegment input, [NotNullWhen(true)] out EntityTagHeaderValue? parsedValue)
```